src/horizon_client/horizon_client.rs

```diff
@@ -243,10 +236,9 @@ mod tests {
             HorizonClient::new("https://horizon-testnet.stellar.org".to_string()).unwrap();
 
         // construct request
-        let mut accounts_request = AccountsRequest::new();
-        accounts_request
-            .set_signer("GDQJUTQYK2MQX2VGDR2FYWLIYAQIEGXTQVTFEMGH2BEWFG4BRUY4CKI7")
-            .set_limit(10);
+        let accounts_request = AccountsRequest::new()
```

Comment:
We can also add unit tests for invalid states by checking that they result in the correct type:
`.is::<T>()`

Reply:
Cannot figure out how to do this, could you point me in the right direction? For example, I get the error
`no method named `is` found for struct `AccountsRequest` in the current scope`

When adding
`assert!(accounts_request.is::<AccountsRequest<NoSponsor, AccountsSigner, NoAsset, NoLiquidityPool>>());`